src/Orchard.Web/Modules/Orchard.Localization/Controllers/AdminController.cs

```diff
@@ -72,6 +73,14 @@ public class AdminController : Controller {
             Services.Notifier.Success(T("Successfully cloned. The translated content was saved as a draft."));
 
             var editorRouteValues = _contentManager.GetItemMetadata(contentItemTranslation).EditorRouteValues;
+            // adds request variables of current controller to the new redirect route 
+            // for example the returnUrl parameter
+            foreach (var key in Request.Form.AllKeys.Where(x=> !x.StartsWith("__") && !editorRouteValues.Keys.Contains(x))) {
+                editorRouteValues.Add(key, Request.Form[key]);
+            }
+            foreach (var key in Request.QueryString.AllKeys.Where(x => !x.StartsWith("__") && !editorRouteValues.Keys.Contains(x))) {
+                editorRouteValues.Add(key, Request.QueryString[key]);
+            }
```

Comment:
In media library the edit button adds a "ReturnUrl" QueryString in order to have a redirection to the caller page when saving/cancelling operation.
The translate link has the prop "Unsafe", so it transforms the request from get + querystring in post + body . So I propagate those parameters through redirection translate (which clone) > edit content, in order to be able to manage the right returnurl (and other parameters if they are there). To restrict the logic to only the "redirecturl" parameter, could be less flexible but more safe. @sebastienros what do you prefer?